mutiny-core/src/federation.rs

```diff
@@ -0,0 +1,612 @@
+use crate::log_debug;
+use crate::{
+    error::MutinyError, logging::MutinyLogger, nodemanager::MutinyInvoice, sql::glue::GlueDB,
+    storage::MutinyStorage, utils::sleep, ActivityItem, HTLCStatus,
+};
+use bitcoin::{
+    hashes::{hex::ToHex, sha256},
+    secp256k1::Secp256k1,
+    util::bip32::ExtendedPrivKey,
+};
+use bitcoin::{
+    util::bip32::{ChildNumber, DerivationPath},
+    Network,
+};
+use fedimint_client::{
+    derivable_secret::DerivableSecret,
+    get_config_from_db,
+    oplog::{OperationLogEntry, UpdateStreamOrOutcome},
+    secret::get_default_client_secret,
+    ClientArc, FederationInfo,
+};
+use fedimint_core::{
+    api::InviteCode,
+    config::FederationId,
+    core::OperationId,
+    task::{MaybeSend, MaybeSync},
+    Amount,
+};
+use fedimint_ln_client::{
+    InternalPayState, LightningClientInit, LightningClientModule, LightningOperationMeta,
+    LightningOperationMetaVariant, LnPayState, LnReceiveState,
+};
+use fedimint_mint_client::MintClientInit;
+use fedimint_wallet_client::WalletClientInit;
+use futures::future::{self};
+use futures_util::{pin_mut, stream, StreamExt};
+use lightning::log_info;
+use lightning::{log_warn, util::logger::Logger};
+use lightning_invoice::Bolt11Invoice;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::fmt::Debug;
+use std::{
+    collections::HashMap,
+    sync::{atomic::AtomicBool, Arc, RwLock},
+};
+
+const FEDIMINT_CLIENT_NONCE: &[u8] = b"Fedimint Client Salt";
```

Comment:
is this supposed to be a fixed value? would be nice to have a comment explaining what this does/is for

Reply:
maybe we should put the name mutiny in it to so we don't clash with other clients? unless we want to?

Reply:
If someone derrives their `m/1'/X`keys, they could use it in theory with other fedimint clients. So it's a feature that might not ever be necessary but could be a bonus for interoperability. I added a comment about it.